### PR TITLE
refactor: ETA calculation

### PR DESCRIPTION
**Pull request by AlexHladin**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain:
ETA calculation refactoring

**What changes did you make? (Give an overview)**
Refactored ETA calculation

**Which issue (if any) does this pull request address?**
[1902](https://github.com/webtorrent/webtorrent-desktop/issues/1902)

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: src/renderer/lib/date.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
```
Comment (alxhotel) [inline src/renderer/lib/date.js:1]: Can you rename this file to `time.js` ?

File: src/renderer/pages/player-page.js [hunk 2]
```diff
@@ -446,16 +447,9 @@ function renderCastScreen (state) {
...
-    const minutes = Math.floor(rawEta / 60) % 60
-    const seconds = Math.floor(rawEta % 60)
+    const etaStr = calculateEta(missing, downloadSpeed)
 
-    const hoursStr = hours ? hours + ' h' : ''
-    const minutesStr = (hours || minutes) ? minutes + ' min' : ''
-    const secondsStr = seconds + ' s'
-
-    return (<span>{hoursStr} {minutesStr} {secondsStr} remaining</span>)
+    return (<span>{etaStr} remaining</span>)
```
Comment (alxhotel) [inline src/renderer/pages/player-page.js:452]:
```suggestion
    return (<span>{etaStr}</span>)
```

File: src/renderer/lib/date.js (hunk 1 above)
Comment (AlexHladin) [reply src/renderer/lib/date.js]: Done

**Comment by alxhotel** [comment]
Thanks @AlexHladin !

Please fix the lint error and I think is ready to merge

**Comment by AlexHladin** [comment]
> Thanks @AlexHladin !
> 
> Please fix the lint error and I think it's ready to merge

Fixed

**Review by alxhotel — APPROVED** [review]

**Pull request merged**